internal/internal.go

```diff
@@ -263,6 +302,71 @@ func (p *envoyExtAuthzGrpcServer) log(ctx context.Context, input interface{}, re
 	return plugin.Log(ctx, info)
 }
 
+// getResponseStatus returns the status of the evaluation.
+// The status is determined by the "status" key in the evaluation
+// result and is represented as a boolean value. If the key is missing, an
+// error will be returned.
+func getResponseStatus(result map[string]interface{}) (int32, error) {
+	status := int32(google_rpc.PERMISSION_DENIED)
+
+	var decision, ok bool
+	var val interface{}
+
+	if val, ok = result["status"]; !ok {
+		return 0, fmt.Errorf("unable to determine evaluation result due to missing \"status\" key")
+	}
+
+	if decision, ok = val.(bool); !ok {
+		return 0, fmt.Errorf("type assertion error")
+	}
+
+	if decision {
+		status = int32(google_rpc.OK)
+	}
+
+	return status, nil
+}
+
+func getResponseHeaders(result map[string]interface{}) ([]*ext_core.HeaderValueOption, error) {
+	var headers map[string]interface{}
+	var val interface{}
+	var ok bool
+
+	responseHeaders := []*ext_core.HeaderValueOption{}
+
+	if val, ok = result["headers"]; !ok {
+		return responseHeaders, nil
+	}
+
+	if headers, ok = val.(map[string]interface{}); !ok {
+		return nil, fmt.Errorf("type assertion error")
+	}
+
+	for key, value := range headers {
+		headerValue := &ext_core.HeaderValue{
+			Key:   key,
+			Value: fmt.Sprintf("%v", value),
+		}
+
+		headerValueOption := &ext_core.HeaderValueOption{
+			Header: headerValue,
+		}
+
+		responseHeaders = append(responseHeaders, headerValueOption)
+	}
+	return responseHeaders, nil
+}
+
+func getResponseBody(result map[string]interface{}) (string, error) {
+	var ok bool
+	var val interface{}
+
+	if val, ok = result["body"]; !ok {
+		return "", nil
+	}
+	return fmt.Sprintf("%v", val), nil
```

Comment:
I think we expect the policy decision to represent the body as a string. Sprintfing is going to break for things like maps/slices/etc.

Reply:
Removed `fmt.Sprintf`